app/src/main/java/com/example/happymeals/database/FireAuth.java

```diff
@@ -0,0 +1,121 @@
+package com.example.happymeals.database;
+
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.example.happymeals.OutputListener;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+
+ */
+
+
+public class FireAuth {
+
+    // create FireAuth to function statically and allow login and register to access it
+    public FirebaseFirestore fireStore;
+    public FirebaseAuth authenticate;
+    static private FireAuth fireauth;
+
+    // getter and setter for FireAuth
+    public static FireAuth getFireauth() {
+        // check if empty than initialize
+        if(fireauth == null) {
+            fireauth = new FireAuth();
+            fireauth.authenticate = FirebaseAuth.getInstance();
+            fireauth.fireStore = FirebaseFirestore.getInstance();
+        }
+        return fireauth;
+    }
+
+    /** 1) Authenticate user Login -- Checks with data to see if userinput is valid and in Firestore
+     * @param: user - email/username
+     * @param: password - user password
+     * @param: listener - checks if UI works as expected
+     */
+
+    public void userLogin(String user, String password, OutputListener listener){
+        fireauth.authenticate.signInWithEmailAndPassword(user, password).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+            @Override
+            // test
+            public void onComplete(@NonNull Task<AuthResult> action) {
+                if (action.isSuccessful()) {
+                    listener.onSuccess();
+                    Log.d("TAG", "User has successfully authenticated");
```

Comment:
"TAG" needs to let the user know where this error message is coming from i.e. which component. TAG is not helpful in this sense. This goes for all the Log's in this class.